BepInEx/Configuration/TomlTypeConverter.cs

```diff
@@ -341,6 +341,50 @@ public static void AddUnityEngineConverters()
 			TomlTypeConverter.AddConverter(typeof(Vector3), jsonConverter);
 			TomlTypeConverter.AddConverter(typeof(Vector4), jsonConverter);
 			TomlTypeConverter.AddConverter(typeof(Quaternion), jsonConverter);
+			
+			TomlTypeConverter.AddConverter(typeof(Rect), new TypeConverter
+			{
+				ConvertToObject = delegate (string s, Type type)
+				{
+					Rect result = default(Rect);
+					if (s != null)
+					{
+						string cleaned = s.Trim(new char[] { '{', '}' }).Replace(" ", "");
+						foreach (string part in cleaned.Split(new char[] { ',' }))
+						{
+							string[] parts = part.Split(new char[] { ':' });
+							float value;
+							if (parts.Length == 2 && float.TryParse(parts[1], out value))
+							{
+								string id = parts[0].Trim(new char[] { '"' });
+								if (id == "x")
+								{
+									result.x = value;
+								}
+								else if (id == "y")
+								{
+									result.y = value;
+								}
+								else if (id == "width" || id == "z")
+								{
+									result.width = value;
+								}
+								else if (id == "height" || id == "w")
+								{
+									result.height = value;
+								}
+							}
+						}
+					}
+
+					return result;
```

Comment:
Maybe move this into a separate function to keep `AddUnityEngineConverters` readable?

Reply:
Did it.